packages/codecs-strings/src/baseX.ts

```diff
@@ -99,7 +99,12 @@ function partitionLeadingZeroes(value: string, zeroCharacter: string): [string,
 
 function getBigIntFromBaseX(value: string, alphabet: string): bigint {
     const base = BigInt(alphabet.length);
-    return [...value].reduce((sum, char) => sum * base + BigInt(alphabet.indexOf(char)), 0n);
+    let sum = 0n;
+    for (const char of value) {
```

Comment:
Strings implement `Iterator` so you can loop over chars without converting them to an array.